youtube_dl/extractor/sverigesradio.py

```diff
@@ -0,0 +1,111 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import compat_str
+from ..utils import (
+    int_or_none,
+    try_get,
+)
+
+
+class SverigesRadioBaseIE(InfoExtractor):
+    _BASE_URL = 'https://sverigesradio.se/sida/playerajax'
+
+    def _extract_audio(self, audio_type, secondary_type, url):
+        audio_id = self._match_id(url)
+        query = {'id': audio_id, 'type': audio_type}
+
+        metadata = self._download_json(
+            self._BASE_URL + '/audiometadata',
+            audio_id, query=query)
+        item = metadata['items'][0]
+
+        formats = []
+
+        audiourl = self._download_json(
+            self._BASE_URL + '/getaudiourl',
+            audio_id, query=dict(query, type=audio_type, quality='high', format='iis'))
+        formats.append({'url': compat_str(audiourl['audioUrl']), 'vcodec': 'none'})
+
+        if secondary_type is not None:
+            secondaryurl = self._download_json(
+                self._BASE_URL + '/getaudiourl',
+                audio_id, fatal=False,
+                query=dict(query, type=secondary_type, quality='high', format='iis'))
+            if secondaryurl and ('audioUrl' in secondaryurl) and secondaryurl['audioUrl'] != "":
+                formats.append({'url': compat_str(secondaryurl['audioUrl']), 'vcodec': 'none'})
```

Comment:
26-29, 32-37 code duplication.

Reply:
Ok, that was why i asked if i should make the code more generic or not. Fix now. The test with a secondary does not work anymore as sveriges radio have changed something. Now i can't find any publication with a secondary url. Keep code to extract secondary anyway?